src/pool/poolClient.ts

```diff
@@ -315,12 +320,13 @@ export function previewRemoval(
 }
 function joinUserState(
   poolState: Pool,
+  stakeData: StakeData,
```

Comment:
tiny nits, you can simplify this type to a bignumber and just pass in `stakeData.cumulativeBalance` at the caller. also as a general principle, i like to add new parameters to the function at the end, to maintain backward compatibilty.  if you put this at the end you could also default this to 0 in the case where there is no cumulativeBalance.